pkg/provider/aws/plugin.go

```diff
@@ -59,7 +61,7 @@ func (a *AWS) CopyConstructEdgesToDag(result *core.ConstructGraph, dag *core.Res
 			}
 			for _, envVar := range construct.EnvironmentVariables {
 				if envVar.Construct == dep.Destination {
-					data.EnvironmentVariables = []core.EnvironmentVariable{envVar}
+					data.EnvironmentVariables = append(data.EnvironmentVariables, envVar)
```

Comment:
This line fixes a bug where we kept overwriting the same env var instead of appending all.